docs/ir/ir-breaking-changes.rst

```diff
@@ -33,3 +33,39 @@ Consequently, if the padding space within a struct is used to store data (e.g. i
     }
 
 We have the same behavior for implicit delete, for example when array of structs is shortened.
+
+ * The order of contract initialization has changed in case of inheritance.
+
+The order used to be:
+ - All state variables are zero-initialized at the beginning.
+ - Evaluate base constructor arguments from most derived to most base contract.
+ - Initialize all state variables in the whole inheritance hierarchy from most base to most derived.
+ - Run the constructor, if present, for all contracts in the linearized hierarchy from most base to most derived.
+
+New order:
+ - All state variables are zero-initialized at the beginning.
+ - Evaluate base constructor arguments from most derived to most base contract.
+ - For every contract in order from most base to most derived in the linearized hierarchy execute:
+     1. State variables are assigned value their initial values, if present at declaration.
```

Comment:
``State variables are assigned value their initial values`` should be ``State variables are assigned their initial values`` - I think that was me not being careful enough with the suggestion :-)... I'll just create a PR fixing it myself...

Reply:
https://github.com/ethereum/solidity/pull/10503